### ISSUE TITLE
Swords of Legends Online (1418100)

### ISSUE THREAD

**Issue by supersteeeeeeeve** [ValveSoftware/Proton#4851]
**Compatibility Report**

- Name of the game with compatibility issues: Swords of Legends Online
- Steam AppID of the game: 1418100 (BETA: 1558640)

**System Information**

- GPU: AMD RX 5600XT
- Driver/LLVM version: Mesa-git 21.2.0_devel.139931.b663c544177-1
- Kernel version: 5.12.4
- Link to full system information report as [Gist](https://gist.github.com/): [Systeminfo](https://gist.github.com/supersteeeeeeeve/34cfbe435d32d06ca093ee0050249690)
- Proton version: Proton Experimental

**I confirm:**

- [ x ] that I haven't found an existing compatibility report for this game.
- [ x ] that I have checked whether there are updates for my system available.

[steam-1558640.log.tar.gz](https://github.com/ValveSoftware/Proton/files/6529947/steam-1558640.log.tar.gz)

**Symptoms**

Swords of Legends:Beta Launcher starts with a white blank window only, 
starting the game directly causes a connection loop, starting it from the 
launcher is required but you cannot click or press any button inside of the launcher.

**Reproduction**

Force the game to use Proton Experimental, start it, you get only a white window.

**Comment by supersteeeeeeeve** [comment]
Here are some screenshots

When you start it normally:
![Bildschirmfoto vom 2021-05-24 18-26-20](https://user-images.githubusercontent.com/40707257/119378298-689f2380-bcbe-11eb-9d6f-f83ad55de8f1.png)

When you rename the game executeable to the launcher the game boots up
![Bildschirmfoto vom 2021-05-24 18-27-40](https://user-images.githubusercontent.com/40707257/119378389-866c8880-bcbe-11eb-9b3b-4a308e1cc5c6.png)

But the game stays on connecting forever.
![Bildschirmfoto vom 2021-05-24 18-28-22](https://user-images.githubusercontent.com/40707257/119378409-8a98a600-bcbe-11eb-9fb2-20fe8582de2c.png)

**Comment by Sizurka** [comment]
By renaming the launcher and adding `serverlist_url=https://serverlist-gujian.gfsrv.net/na.json` to the command line arguments - which appears to be all the launcher does - I am able to connect and get in game.  I haven't yet had a change to test much but it looks to be basically working.

Thanks for the tip about renaming the launcher.

**Comment by supersteeeeeeeve** [comment]
Fired up a Windows 10 VM, to look at the Command in taskmanager about 5 Hours ago, this is the solution, temporary.

Rename the Game to SOLO_Launcher and adding

For EU
`serverlist_url=https://serverlist-gujian.gfsrv.net/eu.json %command%`
For NA
`serverlist_url=https://serverlist-gujian.gfsrv.net/na.json %command%`
as commandline argument fixed it.

Video: https://www.youtube.com/watch?v=A9-2tpqq548

**Comment by lief2** [comment]
This bypass seems to also work if you bought the game directly from Gameforge instead of Steam. You just don't have to add the command line arguments and instead choose your server in the main menu.

I have created a character but have not yet been able to play it because of an infinite loading screen. I guess it's because the EU server is overloaded?

**Comment by supersteeeeeeeve** [comment]
> This bypass seems to also work if you bought the game directly from Gameforge instead of Steam. You just don't have to add the command line arguments and instead choose your server in the main menu.
> 
> I have created a character but have not yet been able to play it because of an infinite loading screen. I guess it's because the EU server is overloaded?

Yes, since we have only one server for EU, guess thats expected.

**Comment by felipetrz** [comment]
Anyone managed to fix the infinite loading screen?

It's not a server load issue, as the game loads fast and runs fine on a windows VM.

**Comment by Sizurka** [comment]
I haven't experienced the infinite loading; I'm able to get into game and play fine.  This was on the NA servers.   Maybe it's AMD related, since this is on an NVidia Titan Xp?

Another possibility is different command line arguments: I just added the `
serverlist_url=https://serverlist-gujian.gfsrv.net/na.json` to the launch options (i.e. omitting the `%command%`, so it's added to the end of the argument list, rather than as a prefix).

[steam-1558640.log.gz](https://github.com/ValveSoftware/Proton/files/6602891/steam-1558640.log.gz)
[System Info](https://gist.github.com/Sizurka/df5d2ce6aa0a3c9c0ea18c7dfe470b7a)

**Comment by Lord-Zer0-ZA** [comment]
I don't know if I should post it here, but if someone stumbles upon this thread for the released (non-beta) version. I got it working by renaming the game to SOLO_launcher and adding (as commandline argument):

For EU (Tested and working)
`serverlist_url=https://serverlist-service-live.gujian-live.external.ing.production.k8s.gfsrv.net/serverlist/eu %command%`

For NA (Not tested)
`serverlist_url=https://serverlist-gujian.gfsrv.net/na_s.json %command%`

Found URLs here: https://steamdb.info/app/1418100/config/

**Comment by sixtysixq** [comment]
Hey so I completely new to pc, only been gaming on a laptop for a year, I have an asus zephyrus g14 (2020 model) ryzan9 4900hs rtx 2060maxq 16gb ram, windows 10 64bit

I have this issue here except I don't get a white screen. 

Straight from downloading and installing the game via steam, the "play" button when pressed will change to "stop" for about 10 to 15 seconds, then revert back to "play" a quick flash of the "preparing to launch SOLO" shows up and disappears. Game doesn't launch. I've tried to launch from multiple places. Desktop icon, start menu icon, steam library, program files and anywhere else there is an exe. Nothing works. I've restarted steam, I've re installed the game multiple times. I'm only using windows defender no other anti-virus software. 

Now I found this solution here, renaming files I have gotten it to launch after renaming files, but I get "internal server error:none" the second it launched and im.forced to quit. 

Verifying file integrity reverts it to not launching again.

I really don't want to refund the game and I've been searching for fixes all over the internet the past 24 hours, I can't seem to.find help, I've contacted support and they said it's a firewall or permissions issue. I dont belive it is because all that looks good.

I know this might not be the place for this comment, but I've got no other options, thanks to anyone who knows what to do.

**Comment by kisak-valve** [comment]
Hello @sixtysixq, Proton is a project which makes it easier to run Windows based gamed on Linux. Since you are on Windows, this compatibility layer is not used on your system.

You might have luck posting your feedback on the game's Steam community forums or https://solo.support.gameforge.com/.

**Comment by kisak-valve** [comment]
While looking for where the game's devs prefer to get feedback, I came across https://steamcommunity.com/app/1418100/discussions/0/3050608452025305446/ which is a fairly strong indicator that 1558640 will be abandoned.

Instead of restarting the discussion on a separate compatibility report, let's repurpose this compatibility report for the non-beta release.

**Comment by sixtysixq** [comment]
Yeah, none of that works, support is a joke and steam community is useless.
Thanks though I didn't notice it was proton. My bad.

**Comment by inverimus** [comment]
The latest proton experimental version works flawlessly now. Launcher also works now.

**Comment by typedefn** [comment]
My OS is the following:

Distributor ID:	Pop
Description:	Pop!_OS 20.10
Release:	20.10

I just got the official launched game from steam.

I had to rename the gujianol.exe to SOLO_Launcher.exe, make a backup of the old SOLO_Launcher.exe...

The following command line argument worked for me:

serverlist_url=https://serverlist-gujian.gfsrv.net/na_s2.json

Note: the serverlist_url contains the na_s2.json(which worked) not just na_s.json, that one didn't work for me.

**Comment by WaCrex** [comment]
For me the launcher seems to work fine, it's just the character "shader?" that gets a bit messed up when standing on fire or on grass in shadows.. [https://youtu.be/YCaTD8lE6hI](https://youtu.be/YCaTD8lE6hI)

**Comment by inverimus** [comment]
> For me the launcher seems to work fine, it's just the character "shader?" that gets a bit messed up when standing on fire or on grass in shadows.. https://youtu.be/YCaTD8lE6hI

I don't have the issue in the video, but I do notice shadows on characters get messed up sometimes. Most often when talking to people in a heavily shadowed area.

**Comment by WaCrex** [comment]
> > For me the launcher seems to work fine, it's just the character "shader?" that gets a bit messed up when standing on fire or on grass in shadows.. https://youtu.be/YCaTD8lE6hI
> 
> I don't have the issue in the video, but I do notice shadows on characters get messed up sometimes. Most often when talking to people in a heavily shadowed area.

Ah, those black bars in the video are due to the recording software, they aren't visible in-game. All I wanted to show with the video was that the game launched just fine and the FPS for my old GTX 1070, with Proton Experimental I didn't need to rename 
any exe-files or adding serverlist_url. As for the "shader?" / "shadow?" issues that I mentioned, they can be seen in these pictures (I had to leave, so I didn't get to upload any pictures until now):

![Screenshot from 2021-07-10 18-31-25](https://user-images.githubusercontent.com/743363/125322612-40d35000-e33e-11eb-928f-f0430d835eef.png)
![Screenshot from 2021-07-10 19-37-52](https://user-images.githubusercontent.com/743363/125322623-4466d700-e33e-11eb-9c3f-c9e46c1eff72.png)
![Screenshot from 2021-07-10 20-48-27](https://user-images.githubusercontent.com/743363/125322633-4761c780-e33e-11eb-922b-8acf5aee6a86.png)

**Comment by 3kinox** [comment]
@WaCrex For me it looks like the same as : https://github.com/doitsujin/dxvk/issues/1784

**Comment by Bragendesh** [comment]
**System Info**
OS: Manjaro Linux
KERNEL: 5.10.49-1-MANJARO
CPU: Intel Core i7-8700K @ 3.70GHz
GPU: AMD AMD TAHITI (DRM 2.50.0, 5.10.49-1-MANJARO, LLVM 12.0.0)
GPU DRIVER: 4.5 Mesa 21.1.4RAM: 16 GB

**Issues**
- Launcher shows up as a white box running Proton 6.3-5
- Launcher works with Experimental, but will not launch the game
- No change when using the rename/launch options workaround

I get the splash screen with the fancy sword logo, but the app then immediately crashes. I did also try on the linux 513 kernel just to see, but no change.

**Logs**
[steam-1418100.log](https://gist.github.com/Bragendesh/bf7a6d346889b931f0186b0578017c76)

**Comment by kisak-valve** [comment]
Hello @Bragendesh, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

I suspect that you're missing some Vulkan driver support for your video card.

**Comment by Bragendesh** [comment]
> Hello @Bragendesh, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.
> 
> I suspect that you're missing some Vulkan driver support for your video card.

So I kept doing some digging in the logfile and just noticed that you sent your reply... I switched to amdgpu and it works now! I had read that my card wasn't supported at first, but it turns out it is and that amdgpu is basically required for Vulkan to work. Thanks for the reply though, because you were spot on.

If anyone else has a related issue with a Radeon card, [this article](https://wiki.archlinux.org/title/AMDGPU#Enable_Southern_Islands_(SI)_and_Sea_Islands_(CIK)_support) helps explain how to switch on arch distros.

**Comment by WaCrex** [comment]
https://gist.github.com/WaCrex/85035c1e730fe9c8838917857253cd04

**Comment by Sojiro84** [comment]
Does anyone else experience fps dips?

I have a consistent dip in fps every 5/6 seconds. I loose about 50% fps for less then a second before it jumps back again to the normal fps. Happens everywhere. Right now staring at a wall and it goes from 100 fps to 50 fps for no reason for a split second.

Makes it really hard to enjoy playing the game with those consistent frame dips.

Currently also installing the game on my crappy work laptop that has Windows to see if the issue is specific to the game or to Linux.

Here you can see the consistent fps spikes (or dips):

![image](https://user-images.githubusercontent.com/13678578/157865598-1088dd38-888d-4a4c-92de-15b906f0aa6a.png)

Edit: the issue is not present with Proton 6.3-8. Seems to happen with Proton 7 and Proton experimental.

**Comment by VoidCnTL** [comment]
I can confirm the performance dips happening with Proton 7/Experimental. Thanks for the tip about 6.3-8 which doesn't seem to have the issue mentioned. With my specs matching the system recommended (AMD R5 1600 and RX 580), I was wondering if the performance issues were related to the game engine until I noticed this issue thread. I'll be happy to help with whatever log/trace is required to debug this issue though.

**Comment by VoidCnTL** [comment]
@kisak-valve Can we have a regression tag on this thread, because I tested the latest Proton 7 with the last dxvk version used in Proton 6.3-8, and the issue persisted so the regression is probably in Wine/Proton.

**Comment by kisak-valve** [comment]
Hello @VoidCnTL, @Sojiro84, can one of you please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression, and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by VoidCnTL** [comment]
[steam-1418100.tar.gz](https://github.com/ValveSoftware/Proton/files/8317240/steam-1418100.tar.gz)
Had to compress the log as it was getting a bit too big. Reached 37 MB within 3 minutes of running the game.
My system info as [gist](https://gist.github.com/VoidCnTL/b5a86c1dbf6c9701405fa69aabaa2f18)

Hope this helps.

EDIT: Just for reference, I used Proton Experimental for this log. @kisak-valve

**Comment by VoidCnTL** [comment]
Hello @kisak-valve sorry for tagging you again, but since this is another issue that I am facing in this game I thought it is only appropriate to bring to your notice also.

Even though I am using Proton 6.3-8, I find that the audio sometimes gets cut off at random moments and the game crashes as soon as I trigger a loading screen after that.

The engine log (attached) suggests that it might be related to FMOD but I'm not sure
[gujianol.log](https://github.com/ValveSoftware/Proton/files/8326136/gujianol.log)
.

FWIW, I am using pipewire and do have both pipewire-pulse and pipewire-alsa packages installed.

**Comment by Sojiro84** [comment]
@kisak-valve here you go!

System info: https://gist.github.com/Sojiro84/f842cb9edd53c7b09ba8d28b427a6221

And the log: [steam-1418100.zip](https://github.com/ValveSoftware/Proton/files/8327097/steam-1418100.zip)

Ran it with Proton Experimental - Bleeding Edge.

**Comment by VoidCnTL** [comment]
Can confirm that the latest Proton-Experimental as on date has fixed the Performance Regression/FPS drops issue. I don't know if the audio issues/crashes are gone too, but they might just be my set up and I'll update the thread in that case later.

Edit: I think I can even say that the performance has improved a lot, at least for me.

Kudos and thanks for the fast response.

**Comment by Sojiro84** [comment]
Same here, fixed now for me as well.

**Comment by HunnnyBunnny** [comment]
I have a pair of error messages upon game start that seem to be part of a problem which is keeping the game from actually initializing.

The first one is as follows:

Line: 1
Char: 5
Error: Expected ';'
Code: 0
URL: About:Blank

The second error is as follows:

Line: 21
Char: 774
Error: Expected Identifier
Code: 0
URL: Abut:Blank

Can anybody help me resolve this?
![Swords of Legends Script Error 1](https://user-images.githubusercontent.com/112135543/186788405-8eb8d014-77d6-41bb-a447-cbaeddcc08dd.jpg)
![Swords of Legends Script Error 2](https://user-images.githubusercontent.com/112135543/186788619-042213a4-fc6a-4fda-8659-0b819213597c.jpg)
